### PR TITLE
add api route for llm query

### PR DESCRIPTION
**Pull request by aldrinjenson**
### Description

- Add API route for LLM query
- read from OPENAI_API_KEY in env var

<hr>

This PR has:
- [x] been tested to ensure log ingestion and log query works.
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added documentation for new or modified features or behaviors.

### CODE REVIEW

**Comment by aldrinjenson** [comment]
I have read the CLA Document and I hereby sign the CLA

File: server/src/handlers/http.rs [hunk 1]
```diff
@@ -229,6 +230,10 @@ pub fn configure_routes(cfg: &mut web::ServiceConfig) {
                         .wrap(DisAllowRootUser),
                 ),
         );
+
+    let llm_query_api =
+        web::scope("/llm").service(web::resource("").route(web::post().to(llm::make_llm_request)));
```
Comment (nitisht) [inline server/src/handlers/http.rs:235]: do we have permissions set up for this API - who can call this API?

Comment (aldrinjenson) [reply server/src/handlers/http.rs:235]: Hi, missed permissions I'm afraid. 
Rn anyone is able to call the API.
Will modify this to need the authorization base64 string being passed in header.

**Review by trueleo — APPROVED** [review]

File: server/src/option.rs [hunk 2]
```diff
@@ -351,6 +356,24 @@ impl Server {
...
+                    .value_name("BOOL")
+                    .required(false)
+                    .default_value("true")
+                    .value_parser(value_parser!(bool))
+                    .help("Disable/Enable sending anonymous user data"),
+            )
+            .arg(
+                Arg::new(Self::OPEN_AI_KEY)
+                    .long(Self::OPEN_AI_KEY)
+                    .env("OPENAI_API_KEY")
```
Comment (nitisht) [inline server/src/option.rs:372]: This should be `P_OPENAI_API_KEY`, so we're consistent with the format we've used till now.

Comment (aldrinjenson) [reply server/src/option.rs:372]: Resolved in https://github.com/parseablehq/parseable/pull/492 👍

**Pull request merged**